acme/http_challenge.go

```diff
@@ -105,17 +34,17 @@ func (s *httpChallenge) startHTTPServer(domain string, token string, keyAuth str
 		// if the domain:port bind failed, fall back to :port bind and try that instead.
 		listener, err = net.Listen("tcp", port)
 		if err != nil {
-			s.end <- err
+			return fmt.Errorf("Could not start HTTP server for challenge -> %v", err)
 		}
 	}
-	// Signal successfull start
-	s.start <- listener
+	defer listener.Close()
 
-	path := "/.well-known/acme-challenge/" + token
+	path := "/.well-known/acme-challenge/" + chlng.Token
 
 	// The handler validates the HOST header and request type.
 	// For validation it then writes the token the server returned with the challenge
-	http.HandleFunc(path, func(w http.ResponseWriter, r *http.Request) {
+	mux := http.NewServeMux()
```

Comment:
Does a mux really solve the same problem here? I used channels not for more "speed" but to be able to explicitly close the listener and wait for it to close. Otherwise there would be racing conditions when solving more then one challenge.